pkgs/development/python-modules/black/default.nix

```diff
@@ -36,6 +40,11 @@ buildPythonPackage rec {
   preCheck = ''
     export PATH="$PATH:$out/bin"
 
+    # Work around https://github.com/psf/black/issues/2105
+    if [[ "${lib.boolToString(stdenv.isDarwin)}" == "true" ]]; then
+      export TMPDIR="/tmp"
```

Comment:
Are you sure this is needed? `$TMPDIR` should always be set.

Reply:
It's the only way I was able to get tests to pass (see the linked issue in the description). The failure was `OSError: AF_UNIX path too long` relating to the temporary directory, which by default is `/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-black-21.5b1.drv-0/tmpov34r68o`.  I know you can also use `/tmp` to access the temp folder on macOS, so I tried this and it worked.

That said maybe there's a better way to resolve this.

Reply:
I see. Then use
```suggestion
      export TMPDIR=$(mktemp -d)
```

Reply:
That creates the same folder basically (`/private/var/folders/1s/gpt7p1x559sb7qjb_4bhq7zc75yv8y/T/nix-build-python3.8-black-21.6b0.drv-0/tmp.59AypUEvaF`) and the same issue happens as a result.

Reply:
Found another issue within the Nix ecosystem that ended up using the same solution: https://github.com/Mic92/sops-nix/pull/15